sdks/java/io/couchbase/src/test/java/org/apache/beam/sdk/io/couchbase/CouchbaseIOTest.java

```diff
@@ -0,0 +1,142 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.testcontainers.couchbase.CouchbaseContainer;
+
+/** Test for {@link CouchbaseIO}. */
+@RunWith(JUnit4.class)
+public class CouchbaseIOTest implements Serializable {
+
+  @Rule public transient TestPipeline pipeline = TestPipeline.create();
+
+  private static CouchbaseContainer couchbase;
+  private static Bucket bucket;
+  private static final int CARRIER_PORT = 11210;
+  private static final int HTTP_PORT = 8091;
+  private static final String BUCKET_NAME = "bucket-name";
+  private static final String USERNAME = "admin";
+  private static final String ADMIN_PWD = "admin-pwd";
+  private static final String BUCKET_PWD = "bucket-pwd";
+  private static final int SAMPLE_SIZE = 100;
+
+  @BeforeClass
+  public static void startCouchbase() {
+    couchbase =
+        new CouchbaseContainer()
+            .withClusterAdmin(USERNAME, ADMIN_PWD)
+            .withNewBucket(
+                DefaultBucketSettings.builder()
+                    .enableFlush(true)
+                    .name(BUCKET_NAME)
+                    .quota(100)
+                    .password(BUCKET_PWD)
+                    .type(BucketType.COUCHBASE)
+                    .build());
+    couchbase.start();
+    bucket = couchbase.getCouchbaseCluster().openBucket(BUCKET_NAME);
+    insertData();
+  }
+
+  @AfterClass
+  public static void stopCouchbase() {
+    if (couchbase.isIndex() && couchbase.isQuery() && couchbase.isPrimaryIndex()) {
+      bucket.query(
+          N1qlQuery.simple(
+              String.format("DELETE FROM `%s`", bucket.name()),
+              N1qlParams.build().consistency(ScanConsistency.STATEMENT_PLUS)));
+    } else {
+      bucket.bucketManager().flush();
+    }
+  }
+
+  @Test
+  public void testEstimatedKeySize() throws Exception {
+    CouchbaseIO.Read read =
+        CouchbaseIO.read()
+            .withHosts(Arrays.asList(couchbase.getContainerIpAddress()))
+            .withHttpPort(couchbase.getMappedPort(HTTP_PORT))
+            .withCarrierPort(couchbase.getMappedPort(CARRIER_PORT))
+            .withBucket(BUCKET_NAME)
+            .withPassword(BUCKET_PWD);
+    CouchbaseIO.CouchbaseSource source = new CouchbaseIO.CouchbaseSource(read);
+    long resultSize = source.getEstimatedSizeBytes(PipelineOptionsFactory.create());
+    TestCase.assertEquals(SAMPLE_SIZE, resultSize);
+  }
+
+  @Test
+  public void testRead() {
+    PCollection<JsonDocument> output =
+        pipeline.apply(
+            CouchbaseIO.read()
+                .withHosts(Arrays.asList(couchbase.getContainerIpAddress()))
+                .withHttpPort(couchbase.getMappedPort(HTTP_PORT))
+                .withCarrierPort(couchbase.getMappedPort(CARRIER_PORT))
+                .withBucket(BUCKET_NAME)
+                .withPassword(BUCKET_PWD));
+
+    PAssert.thatSingleton(output.apply("Count", Count.globally()))
+        .isEqualTo(Long.valueOf(SAMPLE_SIZE));
+
```

Comment:
Please, add the following assert to check that all records were properly read:
`PAssert.that(output).containsInAnyOrder(records);`